packages/rocketchat-lib/client/lib/ChannelActions.js

```diff
@@ -14,16 +25,19 @@ export function hide(type, rid, name) {
 		closeOnConfirm: true,
 		html: false
 	}, async function() {
-		if (['channel', 'group', 'direct'].includes(FlowRouter.getRouteName()) && (Session.get('openedRoom') === rid)) {
-			FlowRouter.go('home');
-		}
+		await handleHideRoom(rid);
+	});
+	return false;
+}
 
-		await call('hideRoom', rid);
-		if (rid === Session.get('openedRoom')) {
-			Session.delete('openedRoom');
+export function hideAllDirect() {
+	Meteor.call('getRoomsByType', 'd', (err, directRooms) => {
```

Comment:
We should probably add a modal here, to prevent accidental hiding. If we are going to have a warning before hiding individual channels, we should definitely have one for hiding them all :) You can have a look at the code a few lines above in hide() for a guide.

Reply:
I think it should be discussed in the #10948  ? What do you think?

Reply:
Recently, an option was added to never show the hiding modal again: #10973 
Perhaps you can hook into that.

Reply:
Ok, I'll add a modal

Reply:
I've added a little demo below